### DISCUSSION TITLE
Docs:  be clear about this being server software

### DISCUSSION THREAD

**Discussion by bburdette** [owncast/owncast#3127] [General Owncast discussion]
From the perspective of someone coming to the owncast website for the first time and trying to figure out what exactly this thing is.

Looking at the owncast quickstart - it says I can just install the software through my package manager.  Ok great.  But isn't owncast a server software that needs to be installed on a server with a domain?  This isn't made clear on the quickstart.  If it is required to have a server, can that be made explicit?  

Posting here as the docs repo has no discussion or issues enabled.

**Comment by gabek** [comment]
It does say it's a "free and open source live video and web chat server", and all server software needs to run on a server. That's like buying a copy of Microsoft Word and it walking you through buying your first computer to install it on. The computer part is implied because you're trying to run software.

Owncast, being self-hosted server software, is unopinionated where and how you run it. Maybe you're running it privately on your home network, in that case you don't need a public server with a public domain name. It's server software like any other, and I'm not going to tell you to go buy a domain to run Owncast because that's not accurate, I don't know what you're doing with it. It's up to you.

I do completely understand that running your own server isn't something that everyone has done before, but there's a certain amount of reasonable scope that makes sense from the documentation point of view. We already have a ton of documentation, and I think it would be out of scope to explain what a server is, how you acquire one, what DNS is, how you buy a domain, etc.

I'm sure this is frustrating, and I'm not trying to belittle your experiences, but I can't possibly document every single use case and all the scenarios just like when you install Apache it doesn't tell you to go buy a domain name first.

**Reply by bburdette** [reply]
It does say that, but not on the quickstart.  And when you go to search results for owncast there is a direct link to quickstart without any its-a-server explanation:

![image](https://github.com/owncast/owncast/assets/157330/f00773ac-33e1-4e83-8044-27a36d5b9577)

A new user doesn't necessarily know that owncast is only a server and not a client also.  What am I getting started with?  A client that can connect to existing servers and stream?  A server that goes in the cloud?  Something decentralized that doesn't need a server?  Its not apparent from the quickstart page.  

I think there are a lot of users who just want to start streaming instantly like they did with rpan, and don't know what they're going to need to do to get going with this.  Providing some direction for this kind of user would be helpful to them and probably helpful for this project as well.  In my opinion.

**Reply by gabek** [reply]
I'm not sure what the solution here is. If somebody refuses to read the web site, and therefore doesn't know what Owncast is, even though it says "server" everywhere (including in that search result screenshot), there's not much more I can do. The quickstart even says specifically "I want to install it myself" or "I want Owncast to be installed for me". That's about as explicit as possible.

Putting on every page "Owncast is server software you need to install yourself" would be extremely redundant. If somebody refuses to read the documentation, then that's their fault. There's only so much I can do.

I unfortunately don't know what rpan, so I can't compare that software to Owncast, but any server software is in the same boat as Owncast. You need the bare minimum of knowledge that software needs to be installed first.

**Reply by bburdette** [reply]
Welp, I consider my message delivered and I don't know that debating it is super productive.  Seems like a great project and look forward to trying it out.  

RPAN was a reddit feature (now defunct) where you could start a stream from the android app.  One could also browse active streams and comment on them, so the viewer experience in that regard was similar to that of owncast.  It was also possible to stream from pro streaming software rather than just the app if you knew what you were doing, but most streamers didn't do that.

**Reply by gabek** [reply]
Keep in mind, I'm not opposed to making things easier for people. I absolutely want that. But I do think there's a minimum expectation of anybody to know what Owncast is by at least reading the minimum of documentation if they want to use it. If you think our documentation is misleading, or the homepage doesn't say what Owncast is and it's actively confusing people, then improvements should be made.
